monitoring/config-map-reference-for-the-cluster-monitoring-operator.adoc

```diff
@@ -196,13 +202,128 @@ Appears in: link:#clustermonitoringconfiguration[ClusterMonitoringConfiguration]
+
+|===
+
+== NodeExporterCollectorConfig
+
+=== Description
+
+The `NodeExporterCollectorConfig` resource defines settings for individual collectors of the `node-exporter` agent.
+
+Appears in: link:#nodeexporterconfig[NodeExporterConfig]
+
+[options="header"]
+|===
+| Property | Type | Description
+|cpufreq|link:#nodeexportercollectorcpufreqconfig[NodeExporterCollectorCpufreqConfig]|Defines the configuration of the `cpufreq` collector, which collects CPU frequency statistics. Disabled by default.
+
+|tcpstat|link:#nodeexportercollectortcpstatconfig[NodeExporterCollectorTcpStatConfig]|Defines the configuration of the `tcpstat` collector, which collects TCP connection statistics. Disabled by default.
+
+|netdev|link:#nodeexportercollectornetdevconfig[NodeExporterCollectorNetDevConfig]|Defines the configuration of the `netdev` collector, which collects network devices statistics. Enabled by default.
+
+|netclass|link:#nodeexportercollectornetclassconfig[NodeExporterCollectorNetClassConfig]|Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default.
+
+|buddyinfo|link:#nodeexportercollectorbuddyinfoconfig[NodeExporterCollectorBuddyInfoConfig]|Defines the configuration of the `buddyinfo` collector, which collects statistics about memory fragmentation from the `node_buddyinfo_blocks` metric. This metric collects data from `/proc/buddyinfo`. Disabled by default.
+
+|===
+
+== NodeExporterCollectorCpufreqConfig
+
+=== Description
+
+The `NodeExporterCollectorCpufreqConfig` resource works as an on/off switch for the `cpufreq` collector of the `node-exporter` agent. By default, the `cpufreq` collector is disabled. Under certain circumstances, enabling the cpufreq collector increases CPU usage on machines with many cores. If you enable this collector and have machines with many cores, monitor your systems closely for excessive CPU usage.
+
+Appears in: link:#nodeexportercollectorconfig[NodeExporterCollectorConfig]
+
+[options="header"]
+|===
+| Property | Type | Description
+|enabled|bool|A Boolean flag that enables or disables the `cpufreq` collector.
+
+|===
+
+== NodeExporterCollectorNetClassConfig
+
+=== Description
+
+The `NodeExporterCollectorNetClassConfig` resource works as an on/off switch for the `netclass` collector of the `node-exporter` agent. By default, the `netclass` collector is enabled. If disabled, these metrics become unavailable: `node_network_info`, `node_network_address_assign_type`, `node_network_carrier`, `node_network_carrier_changes_total`, `node_network_carrier_up_changes_total`, `node_network_carrier_down_changes_total`, `node_network_device_id`, `node_network_dormant`, `node_network_flags`, `node_network_iface_id`, `node_network_iface_link`, `node_network_iface_link_mode`, `node_network_mtu_bytes`, `node_network_name_assign_type`, `node_network_net_dev_group`, `node_network_speed_bytes`, `node_network_transmit_queue_length`, `node_network_protocol_type`.
```

Comment:
Not sure if this can be achieved given that the source of the content is a comment in the source code. But I will add this to my list to investigate for the next release. Thanks!